### ISSUE TITLE
Modify the CelloArray dimension naming

### ISSUE THREAD

**Issue by mabruzzo** [enzo-project/enzo-e#71]
Currently `CelloArray` follows a convention similar to numpy arrays (assuming that the underlying array has C ordering): we enumerate the dimensions in order of increasing access speed. 
- For example, if we have a 3D array, `arr.shape(0)` gives the size of the dimension along which element is slowest and `arr.shape(2)` gives the size of the dimension along which element access is fastest.
- To put it another way, if we initialize an array called `arr` with shape `(a,b,c,d)` then:
    - `arr.shape(0) == a`
    - `arr.shape(1) == b`
    - `arr.shape(2) == c`
    - `arr.shape(3) == d`

In enzo-e, the x-position of cells always varies along the fastest access-order dimension while the z-position of cells always varies along slowest access- order dimension.

This combination of factors frequently leads to this recurring pattern:

```C++
for (int iz = 0; iz < arr.shape(0); iz++) {
  for (int iy = 0; iy < arr.shape(1); iy++) {
    for (int ix = 0; ix < arr.shape(2); ix++) {
      // perform operations involving arr ...
    }
  }
}
```
In other words, the number of elements elements along the x-axis is given by `arr.shape(2)` while the number of elements along the z-axis is given by `arr.shape(0)`. 

This is counter-intuitive, especially because there are numerous places throughout the VL+CT where the dimension of an operation is expected as an argument to a function, with the following mapping:
- `x <--> 0`
- `y <--> 1`
- `z <--> 2`
A similar mapping is also used in some Cello machinery, but is less explicit.

As discussed in PR #62, to resolve this tension, it would probably be best to reverse this number of dimensions. This new ordering would be similar to `AthenaArray` from Athena++ (Note: that in Athena++, the lowest dimension starts at 1, here it would start from `0`)
